specification.md

```diff
@@ -0,0 +1,344 @@
+# $signing_spec
+
+<table>
+<tr><td>$signing_spec
+<tr><td>A signature scheme for software supply chain metadata that avoids canonicalization
+<tr><td>2020-09-28
+<tr><td>Version 0.1.0
+</table>
+
+# Abstract
+
+This document proposes a new signature scheme for use by, among others, the
+in-toto and TUF projects. This signature scheme (a) avoids relying on
+canonicalization for security and (b) reduces the possibility of
+misinterpretation of the payload. The serialized payload is encoded as a string
+and verified by the recipient _before_ deserializing. A backwards compatible
+variant is available.
+
+# Specification
+
+$signing_spec does not rely on Canonical JSON or any other encoding /
+canonicalization. Instead, the producer records the signed bytes
+exactly as signed and the consumer verifies those exact bytes before parsing. In
+addition, the signature now includes an authenticated `payloadType` field
+indicating how to interpret the payload.
+
+```json
+{
+  "payload": "<Base64(SERIALIZED_BODY)>",
+  "payloadType": "<PAYLOAD_TYPE>",
+  "signatures": [{
+    …,
+    "sig": "<Base64(Sign(PAE([PAYLOAD_TYPE, SERIALIZED_BODY])))>"
+  }, …]
+}
+```
+
+where PAE is the
+[PASETO Pre-Authentication Encoding](https://github.com/paragonie/paseto/blob/master/docs/01-Protocol-Versions/Common.md#authentication-padding):
+
+```none
+PAE([type, body]) := le64(2) || le64(len(type)) || type || le64(len(body)) || body
+le64(n) := 64-bit little-endian encoding of `n`, where 0 <= n < 2^63
+```
+
+The PAYLOAD_TYPE is a URI indicating how to interpret SERIALIZED_BODY. It
+encompasses the content type (JSON, CBOR, etc.), the purpose, and the schema
```

Comment:
Should we recommend a canonical content type ? In one of our first discussions about this proposal we agreed that producing the signature over the canonical form has advantages, i.e. it makes signatures reproducible.

Reply:
💯

I'd like to start off this draft as backwards-compatible as possible. We cannot simply break existing clients unless we put this in major versions of tuf and in-toto...

Reply:
I've added Canonical-JSON to the list of options, rather than replace JSON. I wonder if the spec is best served by not taking a stance requiring canonicalization, while we as implementers for in-toto / TUF can use it to get benefits like reproducible signatures. As for backwards compatibility, the document has workflows for it which use canonical JSON for the current document formats. For new documents, the spec is not taking a stance against canonical JSON, it's merely not requiring it. Does that work? :)